src/borneo/operations.py

```diff
@@ -4469,6 +4506,191 @@ def _compute(self):
                     self.get_write_units())
 
 
+class QueryIterableResult(Result):
+    """
+    QueryIterableResult comprises an iterable list of dict instances
+    representing all the query results.
+
+    The shape of the values is based on the schema implied by the query. For
+    example a query such as "SELECT * FROM ..." that returns an intact row will
+    return values that conform to the schema of the table. Projections return
+    instances that conform to the schema implied by the statement. UPDATE
+    queries either return values based on a RETURNING clause or, by default, the
+    number of rows affected by the statement.
+
+    Each iterator from QueryIterableResult will iterate over all results of the
+    query.
+
+    .. code-block:: pycon
+
+        handle = ...
+        request = QueryRequest().set_statement('SELECT * FROM foo')
+        qiresult = handle.query-iterable(request)
+        for row in qiresult:
+            # do something with the result row
+            print(row)
+
+    Modification queries either return values based on a RETURNING clause or, by
+    default, return the number of rows affected by the statement in a
+    dictionary. INSERT queries with no RETURNING clause return a dictionary
+    indicating the number of rows inserted, for example {'NumRowsInserted': 5}.
+    UPDATE queries with no RETURNING clause return a dictionary indicating the
+    number of rows updated, for example {'NumRowsUpdated': 3}. DELETE queries
+    with no RETURNING clause return a dictionary indicating the number of rows
+    deleted, for example {'numRowsDeleted': 2}.
+
+    :versionadded: 5.3.5
+    """
+
+    def __init__(self, request, handle):
+        # type: (QueryRequest, NoSQLHandle) -> None
+        # NoSQLHandle handle
+        super(QueryIterableResult, self).__init__()
+        self.request = request
+        self.handle = handle
+        self.readKB = 0
+        self.readUnits = 0
+        self.writeKB = 0
+
+    def __str__(self):
+        # type: () -> str
+        res = 'QueryIterableResult(' + str(self.request) + ')'
+        return res + '\n'
+
+    def get_read_kb(self):
+        # type: () -> int
+        """
+        Returns the read throughput consumed by this operation, in KBytes. This
```

Comment:
should mention that this (and write_kb) are cumulative, based on the units consumed by all remote calls generated by the iterator

Reply:
done